### ISSUE TITLE
[variable explorer] [enhancement] zooming

### ISSUE THREAD

**Issue by goyou** [spyder-ide/spyder#3165]
Hello,

I am just bringing an idea to eventually enhance the variable explorer...
The colored background is great !
It will become such a wonder if it could be possible to zoom (just reducing the size of the cases).

Using ctrl+mousewheel, if you are able to increase/decrease the size of the (colored) cases you get a fantastic and easy way to navigate and observe the numerical variations within some matrix for example.

I think changing the size of the cases is quite easy, I don't know about the possibility to do it interactively.

Anyway can I configure the size of the cases somewhere ? I cannot find it in the preferences.

**Comment by ccordoba12** [comment]
Interesting idea. However, it doesn't seem easy to implement :-)

**Comment by goyou** [comment]
Ok, thank you.
Is there some way to easily change the size of the cases anyway ?

**Comment by ccordoba12** [comment]
What do you mean by *the size of the cases*?

**Comment by goyou** [comment]
I mean, after opening the window (variable editor?) when double-clicking on a matrix entry inside the variable explorer... A table is displayed, if it is possible to reduce the size of the cases, it is possible to see more of them.

**Comment by Nodd** [comment]
FYI the proposal in #2368 included an overview of the array in a similar way as what is wanted here. It was just an overview with no zooming functionality.
But I think it's the wrong tool to explore the data; it's better to show the data as an image in matplotlib (or equivalent) if you need to explore your data. The table view has very limited functionality for this.

**Comment by goyou** [comment]
Hello back,
While I am looking at the code here :
`spyder/spyderlib/widgets/variableexplorer/arrayeditor.py`
there is :
`class ArrayView(QTableView)`
which contains the method : 
`def resize_to_contents(self)`

... which is (I believe) called when you click the button "adjust" in the variable explorer.
This function will enlarge the columns for the contents to fit inside.

The possibility to reduce the lines and columns size, even if the numbers are not visible anymore, would allow one to overview much more cases, in order to quickly find (thanks to the colored cases which is a wonderful feature) and edit noisy data samples inside a matrix for example.

Scientifically this is cheating the data but in practice it would be a really comfortable tool to get your data exploitable.

The class QTableView is described here : 
http://doc.qt.io/qt-4.8/qtableview.html
It has the attributes "rowHeight", "columnWidth" (which are actually integers, maybe the number of pixels ??) and the functions 'setRowHeight()' and 'setColumnWidth()'.

Maybe it is possible to add one button (for example "manual adjust") just aside "Adjust" in the variable explorer to be able to set rowHeight and columnWidth to the desired values. This seems easy to do... Maybe I could do it myself if I was able to add the button which opens a dialog with two text fields (one where you enter the new value for columnWidth and the other for rowHeight. I think you just have run the functions 'setRowHeight()' and 'setColumnWidth()' with the entered values once the dialog is closed and that is done.

**Comment by ValerioBiscione** [comment]
I'd also really like to have that. Something similar is done anyway when clicking "resize", would it nice to have it more general and allow the user to set the zooming level manually (or - and that would be great - with the mouse scroll!)

**Comment by stragu** [comment]
This is an issue for accessibility. A user with low vision was able to zoom into the console and the script, but not the variable explorer.
Or is the recommended way to do things to change the scaling for the whole application, as suggested in this StackOverflow question?

https://superuser.com/questions/1473131/control-font-size-of-spyder-python-ide-on-everything-except-the-editor-code

Another thing that could help in this case could be the have a setting to control the UI font size, which would affect everything except script and console contents.

**Comment by ccordoba12** [comment]
@stragu, you said:

> This is an issue for accessibility. A user with low vision was able to zoom into the console and the script, but not the variable explorer.

It's not easy to provide a zoom in/zoom out feature for specific panes besides the Editor, IPython console Help.

> Or is the recommended way to do things to change the scaling for the whole application, as suggested in this StackOverflow question?

Yes, that's the recommended way to increase the zoom factor of the entire interface. In Spyder 6.0.7 (our latest version), you can access that option by going to the menu entry `Tools > Preferences`, then the `Application` entry and finally the `Interface` tab.

> Another thing that could help in this case could be the have a setting to control the UI font size, which would affect everything except script and console contents.

That's already possible in 6.0.7 too, again by going to `Tools > Preferences`, then `Appearance` and finally the `Fonts` section. However, after perhaps 16px the interface looks not too good, so I'd recommend you to try the previous solution first.
